source/_integrations/zha.markdown

```diff
@@ -725,10 +725,22 @@ services:
     network_mode: host
 ```
 
-### EZSP error and other log messages
+### Common error codes in ZHA logs and other log messages
 
-#### NCP entered failed state
+#### EZSP error - NCP entered failed state
 
-When you see `NCP entered failed state. Requesting APP controller restart` in logs during normal operation, it indicates a drop in communication between ZHA and the serial interface of the Silabs EmberZNet Zigbee Coordinator.
+When you see `NCP entered failed state. Requesting APP controller restart` in ZHA logs during normal operation, that indicates was or is a drop in communication between ZHA and the serial interface of the Silabs EmberZNet Zigbee Coordinator (EZSP).
 
-The EZSP (EmberZNet Serial Protocol) interface used by Silicon Labs EmberZNet Zigbee Coordinator adapters requires a stable connection to the serial port; therefore, it is not recommended to use a connection over Wi-Fi, WAN, VPN, etc.
+The root cause is lost serial connection and correlated symptoms for reset connections include freezes and device non-responsiveness. The solution is to make sure to always have a stable end-to-end serial connection to the Zigbee Coordinator adapter.
```

Comment:
Nowadays few people use WiFi coordinators. The root cause is 99% of the time a bad addon or a custom integration locking up the Home Assistant event loop or affecting system stability in some way for ~10 seconds.